crates/sargon-uniffi/src/keys_collector/key_derivation_request.rs

```diff
@@ -1,11 +1,30 @@
 use crate::prelude::*;
-use sargon::indexmap::IndexMap;
+use sargon::IndexMap;
 use sargon::{IndexSet, KeyDerivationRequest as InternalKeyDerivationRequest};
 
 /// A collection of derivation paths, on a per-factor-source basis.
 #[derive(Clone, PartialEq, Eq, uniffi::Record)]
 pub struct KeyDerivationRequest {
-    pub per_factor_source: HashMap<FactorSourceIDFromHash, Vec<DerivationPath>>,
+    pub derivation_purpose: DerivationPurpose,
```

Comment:
maybe include one line doc as to how we will use `derivation_purpose` field? something like:

```
/// We include this DerivationPurpose in dispatched use FactorSource requests to host so that UI 
/// can display contextual information as to why the user is prompted to authenticate FactorSource access.
```